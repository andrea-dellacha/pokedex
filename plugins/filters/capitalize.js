import Vue from 'vue'

export default function capitalize (value) {
  if (!value) { return '' }
  value = value.toString()

  return value.charAt(0).toUpperCase() + value.slice(1)
}

Vue.filter('capitalize', capitalize)
