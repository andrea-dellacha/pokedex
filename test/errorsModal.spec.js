import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import ErrorModal from '@/components/errors-modal.vue'
import capitalize from '@/plugins/filters/capitalize'

describe('ErrorModal', () => {
  const localVue = createLocalVue()

  // Filter import
  localVue.filter('capitalize', capitalize)
  const wrapper = shallowMount(ErrorModal, {
    mocks: {
      $store: {
        state: {
          error: {
            error: { title: 'Error', criticality: 1, message: 'an error appened' }
          }
        }
      }
    },
    computed: {
      errorGeneral () {
        return this.$store.state.error.error
      }
    },
    data() {
      return {
        isOpen: true
      }
    },
  })
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should close the modal and clean data', async () => {
    wrapper.vm.isOpen = true
    wrapper.vm.errorsData = { title: 'Error', criticality: 1, message: 'an error appened' }
    expect(wrapper.vm.errorsData).toEqual({ title: 'Error', criticality: 1, message: 'an error appened' })
    expect(wrapper.vm.isOpen).toStrictEqual(true)
    wrapper.vm.closeModal()
    await Vue.nextTick()
    expect(wrapper.vm.isOpen).toStrictEqual(false)
    expect(wrapper.vm.errorsData).toEqual({ title: '', criticality: 1, message: '' })
  })
})
