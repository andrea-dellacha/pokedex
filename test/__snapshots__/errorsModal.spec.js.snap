// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal renders correctly 1`] = `
<div>
  <div
    class="p-disableBg"
  >
    <div
      class="c-modal -border_lvl1"
    >
      <div
        class="m-title -color_lvl1"
      >
        <exclamationmarkround-stub
          class="a-title__icon"
          color="#DF0000"
        />
        
        
      
      </div>
       
      <div
        class="m-msg"
      >
        
        
      
      </div>
       
      <div
        class="m-actions"
      >
        <button
          class="btn -dark -noRightMargin"
        >
          
          OK
        
        </button>
      </div>
    </div>
  </div>
</div>
`;
